config/checkstyle.xml

```diff
@@ -0,0 +1,146 @@
+        <module name="ParameterNumber">
+            <property name="max" value="20"/>
+        </module>
+        <module name="ParenPad"/>
+        <module name="RedundantImport"/>
+        <module name="RightCurly">
+            <property name="severity" value="error"/>
+        </module>
+        <module name="StaticVariableName">
+            <property name="format" value="^[a-z][a-zA-Z0-9]*$"/>
+        </module>
+        <module name="TypeName"/>
+        <module name="TypecastParenPad"/>
+        <module name="UpperEll">
+            <property name="severity" value="error"/>
+        </module>
+        <module name="VisibilityModifier"/>
+        <module name="WhitespaceAfter"/>
+        <module name="WhitespaceAround">
+            <property name="tokens" value="ASSIGN,BAND,BAND_ASSIGN,BOR,BOR_ASSIGN,BSR,BSR_ASSIGN,BXOR,BXOR_ASSIGN,COLON,DIV,DIV_ASSIGN,EQUAL,GE,GT,LAND,LCURLY,LE,LITERAL_ASSERT,LITERAL_CATCH,LITERAL_DO,LITERAL_ELSE,LITERAL_FINALLY,LITERAL_FOR,LITERAL_IF,LITERAL_RETURN,LITERAL_SYNCHRONIZED,LITERAL_TRY,LITERAL_WHILE,LOR,LT,MINUS,MINUS_ASSIGN,MOD,MOD_ASSIGN,NOT_EQUAL,PLUS,PLUS_ASSIGN,QUESTION,RCURLY,SL,SLIST,SL_ASSIGN,SR,SR_ASSIGN,STAR,STAR_ASSIGN,LITERAL_ASSERT,TYPE_EXTENSION_AND"/>
+            <property name="allowEmptyConstructors" value="true"/>
+            <property name="allowEmptyMethods" value="true"/>
+        </module>
+        <module name="MissingDeprecated"/>
+        <module name="MissingOverride"/>
+        <module name="PackageAnnotation"/>
+        <module name="CovariantEquals"/>
+        <module name="DefaultComesLast"/>
+        <module name="ExplicitInitialization"/>
+        <module name="FallThrough"/>
+        <module name="InnerAssignment"/>
+        <module name="StringLiteralEquality"/>
+        <module name="GenericWhitespace"/>
+        <module name="MethodParamPad"/>
+        <module name="FinalClass"/>
+        <module name="MutableException"/>
+        <module name="ArrayTypeStyle">
+            <property name="severity" value="error"/>
+        </module>
+        <module name="Indentation">
+            <property name="basicOffset" value="4"/>
+            <property name="caseIndent" value="4"/>
+            <property name="severity" value="warning"/>
+        </module>
+        <property name="severity" value="warning"/>
+    </module>
+    <module name="SuppressionCommentFilter">
+        <property name="offCommentFormat" value="CSOFF"/>
+        <property name="onCommentFormat" value="CSON"/>
+    </module>
+    <module name="FileLength"/>
+    <module name="FileTabCharacter">
+        <property name="eachLine" value="true"/>
+        <property name="severity" value="error"/>
+    </module>
+    <module name="NewlineAtEndOfFile"/>
+    <module name="SuppressWithNearbyCommentFilter">
+        <property name="commentFormat" value="CSIGNORE"/>
+        <property name="checkFormat" value=".*"/>
+        <property name="checkC" value="false"/>
+    </module>
+</module>
```

Comment:
Checkstyles can be quite subjective - I'm definitely opinionated on the matter.
I'm assuming you ran this against the code base, and since I didn't see much in the way of format changes, can I assume the code generally passes this config?

Reply:
Yes. It is compatible with the code base as it were. You could change it in future should you decide to have specific code style for the repo